src/Uno.Extensions.Navigation.UI/Controls/ActionRequestHandlerBase.cs

```diff
@@ -72,47 +102,75 @@ protected ActionRequestHandlerBase(IRouteResolver routes)
 			var response = await nav.NavigateRouteHintAsync(routeHint, element, data, CancellationToken.None);
 
 			if (binding is not null &&
-			binding.ParentBinding.Mode == BindingMode.TwoWay)
+				binding.ParentBinding.Mode == BindingMode.TwoWay)
 			{
+				if (Logger.IsEnabled(LogLevel.Trace))
+				{
+					Logger.LogTraceMessage("Two way binding, so waiting for result of navigation response");
+				}
 				var resultResponse = response?.AsResultResponse();
 				if (resultResponse is not null)
 				{
 					var result = await resultResponse.UntypedResult;
 					if (result.IsSome(out var resultValue))
 					{
+						if (Logger.IsEnabled(LogLevel.Trace))
+						{
+							Logger.LogTraceMessage("Data returned from navigation so setting Data attached property and updating binding");
+						}
+
 						element.SetData(resultValue);
 						binding.UpdateSource();
 					}
+					else
+					{
+						if (Logger.IsEnabled(LogLevel.Trace))
+						{
+							Logger.LogTraceMessage("No data returned from navigation");
+						}
+					}
 				}
 			}
-		};
+		}
 
 		var handler = eventHandler(action);
 
-		bool subscribed = false;
+		var subscribed = false;
 		if (view.IsLoaded)
 		{
 			subscribed = true;
+			if (Logger.IsEnabled(LogLevel.Trace))
+			{
+				Logger.LogTraceMessage("Subscribing");
+			}
 			subscribe(viewToBind, handler);
 		}
 
-		RoutedEventHandler loadedHandler = (s, e) =>
+		void loadedHandler(object s, RoutedEventArgs e)
```

Comment:
Same here... let's ensure this is following conventional naming with PascalCasing